core/src/main/java/org/infinispan/scattered/impl/ScatteredStateConsumerImpl.java

```diff
@@ -105,6 +105,10 @@ protected void beforeTopologyInstalled(int topologyId, boolean startRebalance, C
       // and ScatteringInterceptor would check according to the new CH.
       for (int segment = 0; segment < newWriteCh.getNumSegments(); ++segment) {
          if (!newWriteCh.isSegmentLocalToNode(rpcManager.getAddress(), segment)) {
+            // We cancel any existing transfers as failed key transfer could move segment
+            // to OWNED state if it fails, concurrently with installing the topology.
```

Comment:
I meant if it's already a "failed key transfer", there's no question of "if it fails". Also, the phrasing makes it sound as if the key transfer installs a topology concurrently with changing the segment's state... it would be better to explain in the comment what could cause a concurrent topology update and why we need to prevent concurrent updates of the segment state.